core/model/asynctask.py

```diff
@@ -0,0 +1,115 @@
+# Async
```

Comment:
Minor: I've been trying to slowly move our code towards being able to run with mypy in strict mode. I think this code will pass those tests with almost no changes, so it would be nice to add this new file to `pyproject.toml` in the `[[tool.mypy.overrides]]` section that enables strict mode.